### DISCUSSION TITLE
Add Checkpoints / Save-points to the game

### DISCUSSION THREAD

**Discussion by jayasuryat** [JayaSuryaT/BallTrap-Issues#5] [Ideas]
Like a lot of other `level-based` games have, **checkpoints** are kind of integral to level-based game.

Which, in this case, is missing. And I would like to change that.

From what I can tell, adding checkpoints to the game is not that big of an engineering task. But there are a few things to consider before getting started, on which I'd like to get as many opinions as possible.

1. How often should these checkpoints be?
2. After starting from a checkpoint, how many initial lives should the player have?
3. How should the game score be calculated?

**Comment by jayasuryat** [comment]
My opinion on this is :

1. I'm kind of torn between having checkpoints on every `5th level` or `10th level`. 
    * Having it on every 10th level would make it a bit more challenging and rewarding for the player. 
    * But on the other side of things (checkpoint on every 5th level), things would get a bit more easier and approachable to a wider group of people.
2. I feel if a player starts from a checkpoint, he should not have a full set of lives, because, then the game would be very easy and there would be no practical reason to start the game afresh. So, **I'd prefer just having a single life if a player starts from a checkpoint**. And again, as usual, he can earn more lives in the game while playing.
3. Right now, the game calculation is pretty simple, it just adds up the percentage of the screen covered as the level progresses. And I feel keeping it like that is enough, players who have started from the start and played till a level will (generally) have a higher score than the players who have started from a checkpoint (as they have skipped a certain number of levels, there wouldn't be any score awarded for that). This creates another incentive for players to start afresh and reach higher levels, instead of relying on checkpoints.

**Reply by bhagath555** [reply]
Hi!! @JayaSuryaT Thank you for the response. 
I am really intrigued by your perspective in developing this game and keen observation into things that are to be added. Almost, all of your suggestions, to address the three major issues before going to the actual development of this feature, are really good. But, when it comes to deciding in what intervals the check/save points be added, I am kind of inclined towards 5,10,15, . . . This may not be a big deal at early levels, but, as the game move to higher levels, this feature perfectly balances between the one life that plays starts with and levels that need to be completed to reach the next checkpoint. Thank you.

**Reply by jayasuryat** [reply]
Hey, @bhagath555,
Thank you for your kind words. Giving meaningful feedback to a project at its earlier stages is the best thing one can do for the project, so thank you so much for that✌️.

Your comment about why to keep it at an interval of `every five levels` makes sense. Fair enough, as the player progresses through different levels, it would become increasingly challenging, and maybe the 10 level interval checkpoint won't cut it when considered with single life initially.

So I'd do this, I'll get started with the 5 level interval checkpoint strategy, then I could do an open POC where you could also play and give feedback. Based on how it feels, we may go ahead and finalize then.
